### PR TITLE
Fix project name cleared when browsing path

### LINKED ISSUE #95038: Project manager erases project name after browsing for project path when creating a new project
### Tested versions

- Reproducible in 4.3rc1 and rc2.
- Not reproducible in 4.3beta3 and earlier.

### System information

Godot v4.3.rc2 - Arch Linux #1 SMP PREEMPT_DYNAMIC Sat, 27 Jul 2024 16:49:55 +0000 - Wayland - GLES3 (Compatibility) - AMD Radeon RX 6600 (radeonsi, navi23, LLVM 18.1.8, DRM 3.57, 6.10.2-arch1-1) - AMD Ryzen 7 5700G with Radeon Graphics (16 Threads)

### Issue description

The project name reverts back to "New Game Project" after browsing for a project path. Also the text is highlighted after closing the browse window but nothing on the "Create New Project" window is focused. It happens on linux with the editor set to either x11 or wayland.

### Steps to reproduce

- Press "Create" button in project manager
- Type new project name
- Press "Browse" button

### Minimal reproduction project (MRP)

N/A

### PR DESCRIPTION
**Pull request by KoBeWi**
Fixes #95038

### CODE REVIEW

File: editor/project_manager/project_dialog.cpp [hunk 1]
```diff
@@ -370,6 +370,8 @@ void ProjectDialog::_browse_project_path() {
 	} else {
 		fdialog_project->set_file_mode(EditorFileDialog::FILE_MODE_OPEN_DIR);
 	}
+
+	hide();
```
Comment (KoBeWi) [inline editor/project_manager/project_dialog.cpp:374]: This fixes the exclusive window error.

**Review by AThousandShips — APPROVED** [review]
Can confirm this works

File: editor/project_manager/project_dialog.cpp (hunk 1 above)
Comment (jsjtxietian) [reply editor/project_manager/project_dialog.cpp:374]: IMHO you should also show the dialog when quit the broswer window?

Comment (KoBeWi) [reply editor/project_manager/project_dialog.cpp:374]: Done.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**